Ladybird/LocationEdit.cpp

```diff
@@ -6,11 +6,77 @@
 
 #include "LocationEdit.h"
 #include "Tab.h"
+#include <AK/URL.h>
+#include <QCoreApplication>
+#include <QPalette>
+#include <QTextLayout>
 
 LocationEdit::LocationEdit(Tab* tab)
     : QLineEdit(tab)
 {
     connect(this, &QLineEdit::returnPressed, this, [&] {
         clearFocus();
     });
+
+    connect(this, &QLineEdit::textChanged, this, [&] {
+        highlight_location();
+    });
+}
+
+void LocationEdit::focusInEvent(QFocusEvent* event)
+{
+    QLineEdit::focusInEvent(event);
+    highlight_location();
+}
+
+void LocationEdit::focusOutEvent(QFocusEvent* event)
+{
+    QLineEdit::focusOutEvent(event);
+    highlight_location();
+}
+
+void LocationEdit::highlight_location()
+{
+    auto url = AK::URL::create_with_url_or_path(text().toStdString().c_str());
```

Comment:
we have a qstring_to_akstring helper method for this

Reply:
Fixed!